Remove custom inventory scripts from the API

AlanCoding opened the pull request:
##### SUMMARY
Connect https://github.com/ansible/awx/issues/7775

##### AWX VERSION
```
18.0.0
```

AlanCoding committed:
Change inventory source factory defaults

Also set source_path to a newly merged test-playbooks script
  this gives similar behavior to the prior default script

AlanCoding committed:
Remove custom inventory script API

AlanCoding committed:
Remove old uses of CustomInventoryScript

some uses are minor references

Fix some test fails

AlanCoding committed:
Remove filter_by_class where it was not working

elyezer commented:
@AlanCoding with these changes I am getting:

```awxkit.exceptions.BadRequest: Bad Request (400) received - {'source_script': ["If 'source' is 'custom', 'source_script' must be provided."]}```

When using awxkit and passing source as `custom` when creating an inventory source. I was expecting to get it as an invalid field with these changes, should we consider that here?

AlanCoding commented:
Yes, sorry, that part just isn't finished yet.

AlanCoding committed:
Remove ForeignKey relations involving CustomInventoryScript

Add migration to carry out corresponding schema change

Add data migration to delete inventory sources
  which are the custom type

Split migration into two files
  data migration touches same rows as schema migration

AlanCoding committed:
Remove source_script field from serializers

Remove some other uses of source_script

AlanCoding committed:
Fix migration bug

Bump migration number

Skip data migration for fresh migrations
  where ContentType for custom inventory scripts
  has not yet been created
    no scripts will exist in this case, so no-op

AlanCoding committed:
Remove source_script from awx collection

AlanCoding committed:
Add command to export custom inventory scripts to tar file

AlanCoding commented on awx/main/tests/functional/analytics/test_counts.py:
```diff
@@ -12,7 +12,6 @@ def test_empty():
         "active_sessions": 0,
         "active_host_count": 0,
         "credential": 0,
-        "custom_inventory_script": 0,
```
@rooftopcellist I'm not yet sure if what I've done here works with analytics. It would be trivial to keep these keys but return a count of 0 every time (so that the schema doesn't change). I just need to figure out how to test and assure that it still works.

AlanCoding committed:
Fix migration order of operations bug

AlanCoding commented:
recheck

AlanCoding commented on awx/main/management/commands/export_custom_scripts.py:
```diff
@@ -0,0 +1,36 @@
+
+                    with open(script_path, 'w') as f:
+                        f.write(cis.script)
+                    os.chmod(script_path, stat.S_IRWXU)
```
@wenottingham Important bit here, this was modeled after your suggestions. But `oct(stat.S_IRWXU) == 0o700`, as opposed to 600, because we need the executable bit set.

What this looks like in the test container:

```
$ tar -tvf my_scripts.tar 
-rwx------ awx/root        912 2021-04-19 12:57 _17__inventory_script_bidstory
...
```

jbradberry reviewed (approved):
Everything that I can think to look for appears covered here.

jbradberry commented:
recheck

jbradberry commented:
regate
